packages/lit-dev-content/site/docs/v3/data/signals.md

```diff
@@ -0,0 +1,408 @@
+  static styles = css`
+    :host {
+      display: block;
+    }
+  `;
+
+  render() {
+    return html`
+      <p>The count is ${count.get()}</p>
+      <button @click=${this.#onClick}>Increment</button>
+    `;
+  }
+
+  #onClick() {
+    count.set(count.get() + 1);
+  }
+}
+```
+
+```html
+<!-- Both of these elements will show the same counter value -->
+<shared-counter></shared-counter>
+<shared-counter></shared-counter>
+```
+
+#### Pinpoint Updates with `watch()`
+
+Signals can be used to acheive more targeted, "pinpoint", DOM updates than a
+full Lit component re-render. To do this we need to watch signals individually
+with the `watch()` directive.
+
+DOM updates performed by the `watch()` directive are batched and still use the
+Lit reactive update lifecycle, but when the current Lit update has only been
+triggered by `watch()` directives, the full template render is skipped and only
+the directives with changed signals have their DOM updated.
+
+This examples is the same as the previous, but only the DOM controlled by the
+`${watch(count)}` binding is updated when the `count` signal changes:
+
+```ts
+import {LitElement, html} from 'lit';
+import {customElement} from 'lit/decorators.js';
+import {SignalWatcher, watch, signal} from '@lit-labs/signals';
+
+const count = signal(0);
+
+@customElement('shared-counter')
+export class SharedCounterComponent extends SignalWatcher(LitElement) {
+  static styles = css`
+    :host {
+      display: block;
+    }
+  `;
+
+  render() {
+    return html`
+      <p>The count is ${watch(count)}</p>
+      <button @click=${this.#onClick}>Increment</button>
+    `;
+  }
+
+  #onClick() {
+    count.set(count.get() + 1);
+  }
+}
+```
+
+In most cases `watch()` will _not_ be a significant performance improvement over
+plain Lit template renders. This is because Lit already only updates the DOM
+for bindings that have changed values.
+
+In this example, the work skipped by a pinpoint update is the identity check
+for the template returned by `render()` and the value check for the `@click`
+binding, both of which are cheap.
+
+The performance savings of `watch()` will tend to scale with the amount of
+template logic and the number of bindings that can be skipped in an update, so
+this will be more significant in templates with lots of logic and bindings.
+
+> [!NOTE]
+>
+> <!-- keep Prettier at bay -->
+>
+> `@lit-labs/signals` does not contain a signal-aware `repeat()` directive yet.
+> Changes to the contents arrays will perform a full render until then.
+
+#### Auto-pinpoint updates with the signals `html` template tag
+
+`@lit-labs/signals` also exports an `html` template tag that will apply the
+`watch()` directive automatically to any signal value passed to a binding.
+
+This can be conveient to avoid the extra characters of the `watch()` directive
+or the `signal.get()` calls required without `watch()`.
+
+If you import `html` from `@lit-labs/signals` instead of from `lit`, you will
```

Comment:
Should be presented earlier? This looks like the most convenient API.

Reply:
Not yet, I'll add some of the downsides to the doc

Reply:
done!

Reply:
Good notice!